nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/TreeCommentManager.java

```diff
@@ -559,4 +578,19 @@ public DocumentRef getCommentedDocumentRef(CoreSession session, DocumentModel co
 
         return commentedDocModel.getRef();
     }
+
+    @Override
+    public boolean hasComments(CoreSession session, DocumentModel document, String author) {
+        String query = String.format( //
+                QUERY_GET_COMMENTS_COUNT_FOR_DOCUMENT_BY_ECM_ANCESTOR_AND_AUTHOR, document.getId(), author);
+        return !session.queryProjection(query, 1, 0).isEmpty();
+    }
+
+    @Override
+    protected boolean hasComments(CoreSession session, DocumentModel document) {
+        String query = String.format( //
+                QUERY_GET_COMMENTS_COUNT_FOR_DOCUMENT_BY_ECM_ANCESTOR, document.getId());
+        return !session.queryProjection(query, 1, 0).isEmpty();
+    }
```

Comment:
Put this method before the other `hasComments` one. Consistency with the interface and `PropertyCommentManager`.